### PR TITLE
#165305723 Rate articles

### PR DESCRIPTION
**Pull request by Philipotieno**
#### What does this PR do?
- Rate articles
#### Description of Task to be completed?
- Implement a 5 start rating of articles
- Get an average rating of an article
#### How should this be manually tested?
- `$ git clone https://github.com/andela/ah-the-immortals-frontend`
- `$ git checkout ft-rate-articles-165305723`
- `$ npm install`
- `$ npm run start:dev`
#### What are the relevant pivotal tracker stories?
[`#165305723`](https://www.pivotaltracker.com/n/projects/2326460/stories/165305723)
#### Screenshots
image
image

### CODE REVIEW

**Review by IssaIan — COMMENTED** [review]
Is there a way of making the stars light in colour before a user submits the rating. The black color is confusing to the user as the user.

**Review by Philipotieno — COMMENTED** [review]
Fixed conflicts

**Comment by Ogutu-Brian** [comment]
> Fix conflicts

Yes @Philipotieno ,
Please fix the conflicts

File: src/components/RateArticles/RateArtcles.jsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    starClick(slug, { rate: nextValue }, rating =>
+      this.setState({
+        rating
+      })
+    );
+  };
+
+  render() {
+    const { article, currentUser } = this.props;
+    return !!currentUser?(
```
Comment (Ogutu-Brian) [inline src/components/RateArticles/RateArtcles.jsx]: Hello @Philipotieno ,
When you negate !!currentUser==currentUser, you need not to use the double negation

File: src/components/RateArticles/RateArtcles.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    const slug = article.slug;
+    starClick(slug, { rate: nextValue }, rating =>
+      this.setState({
+        rating
+      })
+    );
+  };
+
+  render() {
+    const { article, currentUser } = this.props;
```
Comment (Ogutu-Brian) [inline src/components/RateArticles/RateArtcles.jsx:19]: Hello @Philipotieno ,
currentUser should be included in the props validation

File: src/components/RateArticles/RateArtcles.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+class ArticleRating extends React.Component {
+  state = {};
+
+  onStarClick = nextValue => {
+    const { article, starClick } = this.props;
+    const slug = article.slug;
+    starClick(slug, { rate: nextValue }, rating =>
+      this.setState({
+        rating
```
Comment (Ogutu-Brian) [inline src/components/RateArticles/RateArtcles.jsx]: Hello @Philipotieno ,
You are not using the state element `rating` so you could do away with it. On matters that, the method:
```
  onStarClick = nextValue => {
    const { article, starClick } = this.props;
    const slug = article.slug;
    starClick(slug, { rate: nextValue }, rating =>
      this.setState({
        rating
      })
    );
  };
```
In `RateArticles.jsx` could not be achieving its intended purpose

**Review by Ogutu-Brian — CHANGES_REQUESTED** [review]

**Comment by Philipotieno** [comment]
@Ogutu-Brian thanks for that I
 am fixing all the hound Issues

**Review by Ogutu-Brian — APPROVED** [review]

**Pull request merged**